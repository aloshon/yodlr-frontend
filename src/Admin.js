import axios from "axios";
import React, {useEffect, useState} from "react";
import Container from "react-bootstrap/Container";
import Button from "react-bootstrap/Button";
import User from "./User";
import {useHistory} from "react-router-dom";

/**
 * Admin fetches list of all users from api and lists them.
 * Display loading text while fetching users.
 */
const Admin = () => {
    const [users, setUsers] = useState(null);
    const history = useHistory();
    let user = JSON.parse(localStorage.getItem("user"));
    
    useEffect(() => {
        async function getUsers(){
            const {data} = await axios.get("http://localhost:3001/users")
            setUsers(data);
        }
        getUsers();
    }, []);
    
    // user is not signed in
    if(!user) history.push("/");
    
    if(!users) return <h1>Loading...</h1>;
    
    return (
        <Container fluid>
            <h1 style={{ margin: "20px", fontSize: "60px"}}>Admin Page</h1>
            {users.map(u => (<User 
                key={u.id}
                id={u.id}
                firstName={u.firstName}
                lastName={u.lastName}
                email={u.email}
                state={u.state}
                isAdmin={u.isAdmin}
                />
            ))}
            <Button 
            variant="secondary"
            style={{position: "fixed", left: 0, bottom: 0}}
            onClick={() => history.push("/")}>
                Back
            </Button>
        </Container>
    )
}

export default Admin