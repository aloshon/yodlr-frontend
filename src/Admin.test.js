import { render } from '@testing-library/react';
import Admin from './Admin';
import {Router} from 'react-router-dom';
import { createMemoryHistory } from 'history';
import StateContext from "./StateContext";

describe("Admin should be rendered", () => {
    const history = createMemoryHistory();
    const {asFragment, getByText} = render(
        
        <Router history={history}>
            <StateContext.Provider value={{updateState: jest.fn()}}>
                <Admin />
            </StateContext.Provider>
        </Router>
    );

    // users are not fetched from api, so test for loading screen
    test('Load admin page loading text', () => {
        const loadingText = getByText("Loading...");
        expect(loadingText).toBeInTheDocument();
    });
    
    test("matches snapshot", () => {
        expect(asFragment()).toMatchSnapshot();
    });
});
