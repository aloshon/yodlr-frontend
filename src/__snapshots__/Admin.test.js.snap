// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Admin should be rendered matches snapshot 1`] = `<DocumentFragment />`;
